framework/include/userobject/BatchMaterial.h

```diff
@@ -0,0 +1,276 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ElementUserObject.h"
+
+#include "libmesh/int_range.h"
+
+#include <tuple>
+#include <string>
+#include <vector>
+#include <utility>
+
+namespace BatchMaterialUtils
+{
+
+// type wrappers
+template <typename T>
+struct GatherMatProp
+{
+  typedef T type;
+  typedef const MaterialProperty<T> ref;
```

Comment:
why call this ref? This immediately makes me think of a reference, and where this typedef is used has nothing to do with references